sabnzbd/nzbqueue.py

```diff
@@ -756,11 +756,7 @@ def register_article(self, article: Article, success: bool = True):
 
         # Write data if file is done or at trigger time
         # Skip if the file is already queued, since all available articles will then be written
-        if file_done or (
-            articles_left
-            and (articles_left % DIRECT_WRITE_TRIGGER) == 0
-            and not sabnzbd.Assembler.partial_nzf_in_queue(nzf)
-        ):
+        if file_done or (articles_left and (articles_left % DIRECT_WRITE_TRIGGER) == 0):
```

Comment:
I wonder how this will play out, if it doesn't just freeze because the queue is full of first articles from the other PR.
For example a huge season pack with 500 files would block the assembler queue for quite a while.
Not really related to this PR, just thought of it while seeing this change.
Will have to test the effects on slow devices with small cache, but with fast internet..

Reply:
I think they should be written quickly enough to mostly keep up because it's only a single part and it's the first in the article list. Writing them one at a time in the assembler instead of concurrently through the decoder might even be an advantage, at least on spinning disks. I guess we'll find out.

Reply:
If it does cause problems then an alternative could be to add a simple accounting of how many articles the queue actually contains. For instance, first article would only add 1, the trigger would add the specified number and done could add trigger / 2. The numbers would be included in the queued elements and subtracted by the assembler. An better alternative but less simple would be to have sparse files and force some flushing if the cache is getting full.